cmd/edit/addon/cmd.go

```diff
@@ -206,9 +206,13 @@ func run(cmd *cobra.Command, argv []string) {
 			if addOnInstallationParam != nil {
 				dflt = addOnInstallationParam.Value()
 			}
-
+			question := param.Name()
+			// add a prompt to question name to indicate if the boolean param is required to be true
+			if param.ValueType() == "boolean" && param.Validation() == "^true$" {
+				question = fmt.Sprintf("%s (required to be true)", param.Name())
+			}
```

Comment:
What's the use-case for having a parameter whose only possible value is `true`?

Reply:
Its basically, agree to terms and conditions box. The RHOAM team are currently using it https://gitlab.cee.redhat.com/service/managed-tenants/-/blob/main/addons/rhoams/metadata/stage/addon.yaml#L105-112

It translates well in the UI, as it is flagged as required and you can not proceed until you select it, but in the CLI and API it seems to get lost in translation. As it is not picked up until validation is done. 

This change will add the prompt that it needs to be true, allowing the user to drill down `?` for the description for more info if they wish. I am also adding a `validation_error` to the addon metadata https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/3259, so the addon maintainer can specify a better error message if the validation fails. My thinking is having a clearer validation error message instead of saying `it must match <<regular expression>>` might negate the need for having the prompt.

Reply:
okay, this makes sense. Now, isn't having a Boolean field marked as Required the same as having the Validation `^true$`? Due to how the data is processed, you can't really mark a boolean as `false` if it's required (I guess unless we take the value as a pointer and consider it a tri-state).

In any case, I think this is fine.

Reply:
Not really, because `Required` only checks if a value is passed, and it doesn't check the actual values value. In the UI, there is no way to not send a boolean value, as the checkbox defaults to false. This is why we are supporting this piece of regex validation to allow for a boolean that has to be true, and mimic a confirmation box in the UI. Its just unfortunate it does not really translate too good to the CLI or API.  

I guess ideally, we probably should have a new parameter type, something like `confirmation` that is a required boolean that has to be true.

Reply:
okay I see - right, we take parameters all as strings and then parse/validate in the API :facepalm: 

Alright, let's not rewrite all this API logic for the sake of a CLI prompt :)